servers/src/main/java/tachyon/master/next/filesystem/meta/InodeTree.java

```diff
@@ -231,6 +231,27 @@ public Inode createPath(TachyonURI path, long blockSizeBytes, boolean recursive,
   }
 
   /**
+   * Returns the first(shortest) path prefix that is nonexistent in the inode tree.
+   *
+   * @param path The path to check whether its prefixes are in the inode tree
+   * @return The first nonexistent path prefix, or null if the path is in the tree
+   * @throws InvalidPathException when the path is invalid
+   */
+  public TachyonURI firstNonexistentPathPrefix(TachyonURI path) throws InvalidPathException {
```

Comment:
Instead of having to introduce this method, I think createPath() should just return a list of inodes, which is basically the ordered list of inodes which have been created by the createPath(). That means, the last inode in the list is the path requested, and the first one in that list would be equivalent to what is returned by this.

This way, the traversal is done only once. Thanks!